datafusion/proto/proto/datafusion.proto

```diff
@@ -714,6 +714,9 @@ message Union{
 }
 
 message ScalarListValue{
+    // encode null explicitly to distinguish a list with a null value
+    // from a list with no values)
+    bool is_null = 3;
```

Comment:
I believe much of the complexity of the code to serialize List values was related to trying to figure out NULL values. Encoding it explicitly makes the code much simpler

Reply:
Why is a null even being encoded at this level, and not using the null_value support at the ScalarValue level?

Reply:
The short answer is I don't know -- the longer answer is that the `null_value` support at the scalar level requires a strange mirror type (`PrimitiveScalarValue`) which doesn't quite have the same support for types as the normal `DataType` -- I am working in the background to remove  that structure in https://github.com/apache/arrow-datafusion/pull/3612 and I will also try to remove this explict null coding as well